spikeinterface/extractors/nwbextractors.py

```diff
@@ -187,7 +187,9 @@ def __init__(
             channel_ids = self._es.electrodes["channel_name"][:].astype("str")
         else:
             channel_ids = [self._es.electrodes.table.id[x] for x in self._es.electrodes.data]
-
+        num_traces = self._es.data.shape[1] 
+        channel_ids = channel_ids[0:num_traces]
```

Comment:
This seems a very too easy fix! how can we be sure that the data correspond to the biginning of channel map ?

Reply:
We can't unfortunately..@luiztauffer are there other columns in the electrode tables that could tell the electrodes apart?

Reply:
I sent a better one now, that I think solves this

Reply:
Cool better otherwise we need a big big warning!
But thother solution look like a better way even I do not understand it.